src/types/_app.ts

```diff
@@ -1,20 +1,32 @@
+import { FC } from 'react'
+import type { NextPage } from 'next'
 import type { AppProps, AppContext } from 'next/app'
+import { CurrentContentType } from 'contexts'
 
-import type { PageWithLayout } from './layouts'
+type CustomAppComponent<
+	PageProps = Record<string, unknown>,
+	InitialProps = PageProps
+> = NextPage<PageProps, InitialProps> & {
+	contentType?: CurrentContentType
+	layout?: FC
+}
```

Comment:
🤔 I don't think `CustomAppComponent` makes sense as a name in the context of its usage, which is a type for any of our page components. What do you think about `CustomPageComponent`? or `DeveloperPageComponent`?

Reply:
Ahhh you're right! I'll think on this a little more

Reply:
Updated to `CustomPageComponent` in https://github.com/hashicorp/dev-portal/pull/953/commits/0f071d57e59642b1135acb79dde4097f79b533d1 ✅